services/motion/builtin/move_request.go

```diff
@@ -66,7 +66,9 @@ type moveRequest struct {
 	kinematicBase     kinematicbase.KinematicBase
 	obstacleDetectors map[vision.Service][]resource.Name
 	replanCostFactor  float64
-	fsService         framesystem.Service
+	// TODO(RSDK-8683): remove alreadyAtGoalCheck and put it in the motionplan package
+	alreadyAtGoalCheck func(basePose spatialmath.Pose) *state.ExecuteResponse
```

Comment:
Lets change this to return the struct rather than a pointer to it

Reply:
I also don't like this name can we call it successCriteria or successFn or something similar?

Reply:
successCriteria or successFn are both vague. I prefer alreadyAtGoalCheck since it is descriptive.

Reply:
atGoalCheck is fine